include/dgl/aten/coo.h

```diff
@@ -133,6 +140,18 @@ struct COOMatrix {
         col_sorted);
   }
 
+  /** @brief Return a copy of this matrix in pinned (page-locked) memory. */
+  inline COOMatrix PinMemory() {
+    if (is_pinned) return *this;
+    auto new_coo = COOMatrix(
+        num_rows, num_cols, row.PinMemory(), col.PinMemory(),
+        aten::IsNullArray(data) ? data : data.PinMemory(), row_sorted,
+        col_sorted);
+    CHECK(new_coo.is_pinned)
+        << "The new allocated COOMatrix is not pinned correctly";
```

Comment:
Is it possible to display the reason why it fails?

Reply:
I think it can only fail when `data.PinMemory().is_pinned()` is false  (or `row`, `col`). Here, unless I scan every ndarray to print out the one (ones) that fails, it can hardly print further useful info for developers and users...

Reply:
OK NVM.